src/pluginextension.h

```diff
@@ -0,0 +1,76 @@
+/*
+ *      pluginextension.h - this file is part of Geany, a fast and lightweight IDE
+ *
+ *      Copyright 2023 The Geany contributors
+ *
+ *      This program is free software; you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation; either version 2 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License along
+ *      with this program; if not, write to the Free Software Foundation, Inc.,
+ *      51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef GEANY_PLUGIN_EXTENSION_H
+#define GEANY_PLUGIN_EXTENSION_H 1
+
+#include "document.h"
+
+/* Temporary define so the plugin can check whether it's compiled against
+ * Geany with plugin extension support or not */
+#define HAVE_GEANY_PLUGIN_EXTENSION 1
+
+G_BEGIN_DECLS
+
+
+typedef struct {
+	gboolean (*autocomplete_provided)(GeanyDocument *doc);
+	void (*autocomplete_perform)(GeanyDocument *doc, gboolean force);
+
+	gboolean (*calltips_provided)(GeanyDocument *doc);
+	void (*calltips_show)(GeanyDocument *doc, gboolean force);
+
+	gboolean (*goto_provided)(GeanyDocument *doc);
+	void (*goto_perform)(GeanyDocument *doc, gint pos, gboolean definition);
+
+	gboolean (*doc_symbols_provided)(GeanyDocument *doc);
+	GPtrArray *(*doc_symbols_get)(GeanyDocument *doc);
+
+	gboolean (*symbol_highlight_provided)(GeanyDocument *doc);
+
+	gchar _dummy[1024];
```

Comment:
> Maybe it'd be safer to have e.g. void (*_dummy[42])(void);

Yeah, it's better. Basically the idea was to pass "infinity" here from the practical point of view as I don't expect this will grow very fast.

On the  other had if it grows slowly, say 5 calls per release and we don't update the padding when adding new functions and keep the struct growing, one would have to still use the 7-8-release-old plugin binary (with that 42 padding) to run into problems.